controllers/goharbor/configuration/reconciler.go

```diff
@@ -0,0 +1,258 @@
+package configuration
```

Comment:
Really not sure about that one.
Talked about this to @steven-zou some weeks ago.
Using a `ConfigMap` to configure Harbor Custom Resource deny the possibility to have a schema validation.
So the validation (even the simpliest) should be done through a webhook.
Moreover config value are no more strongly typed.
There is no automatic client generation, documentation, ...
There is no built list of expected values.

TLDR; I suggest to move all keys expected in this config to the right CRD (Harbor Core? I am not even sure that it is the core CRD or not).

Reply:
@holyhope 

CRD may be a better way, but NOT 100% perfect. I have created an issue to track the work.https://github.com/goharbor/harbor-operator/issues/444.

So far, as we're focusing on fixing the stability related issues, we don't have enough resources to do the enhancements. Let's keep it as it is now (at least it's usable). If we release resources later, then we can start to have a try of doing enhancement.